hi-ml-histopathology/testhisto/testhisto/utils/test_report_utils.py

```diff
@@ -10,7 +10,76 @@
 
 from health_azure.utils import download_file_if_necessary
 from histopathology.utils.report_utils import (collect_crossval_metrics, collect_crossval_outputs,
-                                               get_best_epoch_metrics, get_best_epochs, get_crossval_metrics_table)
+                                               crossval_runs_have_val_and_test_outputs, get_best_epoch_metrics,
+                                               get_best_epochs, get_crossval_metrics_table,
+                                               run_has_val_and_test_outputs)
+
+
+def test_run_has_val_and_test_outputs() -> None:
+    outputs_folder = "outputs/"
+    outputs_filename = "test_output.csv"
+    legacy_outputs_filename = outputs_folder + outputs_filename
+    val_outputs_filename = outputs_folder + "val/" + outputs_filename
+    test_outputs_filename = outputs_folder + "test/" + outputs_filename
+    arbitrary_filename = outputs_filename + "some_other_file"
+
+    run = MagicMock(display_name="mock run", id="abc123")
```

Comment:
For extra certainty, you could try creating a real Run objects with `create_aml_run_object` and uploading (or not) the expected files, then calling `run_has_val_and_test_outputs` on those

Reply:
I'm not sure this function is important enough to warrant that... There are other parts with little coverage where this approach makes more sense. We'll get around to those in time :)